File: 153-FindMinimumInRotatedSortedArray/153-FindMinimumInRotatedSortedArray.py
# Last updated: 07/04/2026, 11:40:53
class Solution:
    def findMin(self, nums: List[int]) -> int:
        min_val = 1e5
        for i, num in enumerate(nums):
            if num < min_val:
                min_val = num
            if i > 0 and num < nums[i-1]:
                break
        return min_val
           

File: 153-FindMinimumInRotatedSortedArray/153-FindMinimumInRotatedSortedArray.cpp
// Last updated: 07/04/2026, 11:40:47
class Solution {
public:
    int findMin(vector<int>& nums) {
        

        // declaration 
        int n = nums.size();
        int low = 0 ; 
        int high = n-1;
        int mid;
        int res = nums[0] ;



        while (low <= high) {

              mid = (low + high)/2 ;
              res = min({res ,nums[mid],nums[low],nums[high]});

              if (nums[low] <= nums[high]){

                return res ;
              }
              if (nums[mid] > nums[low] ) {
                low  = mid +1 ;
              }
              else{
                high = mid-1;

              }
        }

    return res ;




    }
};